main_test.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// parseProjectDir
+func Test_GivenCustomDirProvided_WhenParseProjectDirCalled_ThenExpectCustomDir(t *testing.T) {
+	// Given
+	expectedDir := "/customDir"
+	customOptions := []string{"--project-directory", expectedDir}
+
+	// When
+	acutalProjectDir := parseProjectDir("/originalDir", customOptions)
+
+	//Then
+	assert.Equal(t, expectedDir, acutalProjectDir)
+}
+
+func Test_GivenCustomDirNotProvided_WhenParseProjectDirCalled_ThenExpectOriginalDir(t *testing.T) {
+	// Given
+	expectedDir := "/originalDir"
+	customOptions := []string{"--someparam", `"some value"`}
+
+	// When
+	acutalProjectDir := parseProjectDir(expectedDir, customOptions)
+
+	//Then
+	assert.Equal(t, expectedDir, acutalProjectDir)
+}
+
+// parseCarthageOptions
+func Test_WhenParseCarthageOptionsCalled_ThenExpectCorrectValue(t *testing.T) {
+	// Given
+	expectedOpts := []string{"--parameter", "value"}
```

Comment:
If under real circumstances this can contain URLs, I'd add that in the test case as well, since that could be more error prone to splitting

Reply:
I don't believe that any URLs would be passed here as the `xcconfig` input is passed as an env variable to the command.

Reply:
Right, I confused it with the xcconfig path parsing 🙈